### DISCUSSION TITLE
Actividad 4 - Grupo 2

### DISCUSSION THREAD

**Discussion by BenjaminPeres596** [frlp-utn-ingsoft/catedra#164] [Show and tell]
https://github.com/JuaniLezcano/vetsoft

**Comment by ivanduva** [comment]
Buenas! Les puedo pedir el link a donde lo deployaron?

**Reply by BenjaminPeres596** [reply]
https://vetsoft-g2.onrender.com/

**Reply by ivanduva** [reply]
Buenas,
Les vamos a pedir algunos cambios para cerrar el trabajo:

- Sobre la validación de que el teléfono empiece con 54, no aparece el error en el form y los tests que se agregaron no prueban que salte la error tampoco. Deberían arreglar ambas cosas.
- Sobre la validación del email, se agregó un test de unidad que prueba que no se modifique el campo cuando se agrega algo inválido. Ese enfoque no es del todo relevante porque puede fallar por mil cosas. Deberían testear que salte el error. Por otro lado también les pedimos que agreguen uno de integración y otro e2e.
- Luego de hacer todo, saquen una release. Piensen bien que número de versión correspondería y además fijense que cuando documentan la relase están hablando en primera persona, lo cual no se suele hacer. La mejor documentada es esta: https://github.com/JuaniLezcano/vetsoft/releases/tag/v1.0.0. Una release no es lo mismo que un PR, pueden incluirse cambios introducidos por varias personas.

Recuerden que los cambios deben ser realizados por las personas que se encargaron de cada issue.

Por otro lado, seguimos viendo problemas en las reviews. Más allá de la poca interacción, hay algunas malas prácticas como por ejemplo dejar comentarios para más adelante. Eso no es algo que se suela hacer, generalmente cuando se hacen comentarios es para mejorar el PR y se solucionan ahí mismo. Tengan en cuenta esto en la corrección.

**La última fecha límite de recuperatorio, tanto para esta como para próximas correcciones, será el último día de cursada antes de las vacaciones: Lunes 8 de Julio 23:59 hs.**

Saludos

**Reply by ValentinMilocco** [reply]
Hola Ivan te hago una consulta con respecto a la correccion sobre la validacion del telefono, la parte de que los test que se agregaron no prueban que salte el error, te referis a que los test no estan respondiendo de forma adecuada al hecho de que no aparezca el error en el form?

**Reply by ValentinMilocco** [reply]
A su vez se me genero otra consulta de como subir la solucion, debo hacer un push con merge directo? o debo hacer un pr?

**Reply by ivanduva** [reply]
Son dos cosas, por un lado el error no aparece en el form y por otro los tests de unidad e integración no prueban que aparezca el error que ustedes definieron en la función `validate_client`.
Tienen que trabajar como lo vienen haciendo, con PRs y revisión de los demás integrantes.
Saludos

**Reply by ValentinMilocco** [reply]
Hola Ivan, ya sacamos la nueva Release.

**Reply by ivanduva** [reply]
Buenas,
Hay algunos problemas. Para cerrar el trabajo le vamos a pedir a Valentín lo siguiente:

- En el test `test_validation_invalid_email ` prueban que salte el error con un texto que no tiene formato de email. La idea es que prueben con un formato de mail pero que no termine en "vetsoft.com", que es la validación nueva que ustedes agregaron. Agregá otro test que pruebe eso.
- El test e2e que agregaron no hace ningún expect más allá de este: `expect(self.page.get_by_role("form")).to_be_visible()` que prueba que el form esté visible. Por lo tanto no prueba nada. Debería probar que salte el error en el form.
- Luego de arreglar esto, sacá una release nueva.

Los demás integrantes deberán hacer review de lo que está haciendo Valentín.

Saludos

**Reply by PeresValentin** [reply]
Hola Iván! Ahí acomode los test como me pediste, ya esta la nueva Release.
Saludos!

**Comment by ivanduva** [comment]
Perfecto, el trabajo integrador queda aprobado.

Saludos!
